### PR TITLE
Add POC fuzzer for Skia

### PR DESCRIPTION
**Pull request by kjlubick**
This gets Skia to compile with libfuzzer and ports one of the stable, existing fuzzers.  If this goes well, more will follow.

### CODE REVIEW

File: projects/skia/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+# checkout all sources needed to build your project
+RUN git clone https://skia.googlesource.com/skia.git
+
+# current directory for build script
+WORKDIR skia
+
+RUN python tools/git-sync-deps
+
+COPY build.sh $SRC/
```
Comment (inferno-chromium) [inline projects/skia/Dockerfile]: Can you add a newline.

File: projects/skia/build.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# limitations under the License.
+#
+################################################################################
+
+export CXXFLAGS_ARR=`echo $CXXFLAGS | sed -e "s/\s/\",\"/g"`
+$SRC/depot_tools/gn gen out/Fuzz --args='cc="'$CC'" cxx="'$CXX'" is_debug=false extra_cflags=["'"$CXXFLAGS_ARR"'","-DIS_FUZZING"] skia_use_system_freetype2=false skia_use_fontconfig=false skia_enable_gpu=false extra_ldflags=["-L'$LIB_FUZZING_ENGINE'", "'"$CXXFLAGS_ARR"'"]'
+
+
+
+$SRC/depot_tools/ninja -C out/Fuzz fuzz
```
Comment (inferno-chromium) [inline projects/skia/build.sh]: Can you add a newline.

File: projects/skia/build.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+export CXXFLAGS_ARR=`echo $CXXFLAGS | sed -e "s/\s/\",\"/g"`
+$SRC/depot_tools/gn gen out/Fuzz --args='cc="'$CC'" cxx="'$CXX'" is_debug=false extra_cflags=["'"$CXXFLAGS_ARR"'","-DIS_FUZZING"] skia_use_system_freetype2=false skia_use_fontconfig=false skia_enable_gpu=false extra_ldflags=["-L'$LIB_FUZZING_ENGINE'", "'"$CXXFLAGS_ARR"'"]'
+
```
Comment (inferno-chromium) [inline projects/skia/build.sh]: Can you condense these three empty lines into one.

File: projects/skia/build.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+export CXXFLAGS_ARR=`echo $CXXFLAGS | sed -e "s/\s/\",\"/g"`
+$SRC/depot_tools/gn gen out/Fuzz --args='cc="'$CC'" cxx="'$CXX'" is_debug=false extra_cflags=["'"$CXXFLAGS_ARR"'","-DIS_FUZZING"] skia_use_system_freetype2=false skia_use_fontconfig=false skia_enable_gpu=false extra_ldflags=["-L'$LIB_FUZZING_ENGINE'", "'"$CXXFLAGS_ARR"'"]'
```
Comment (inferno-chromium) [inline projects/skia/build.sh]: Can you break this across multiple lines for readability.

File: projects/skia/build.sh [hunk 5]
```diff
@@ -16,8 +16,10 @@
...
-$SRC/depot_tools/gn gen out/Fuzz --args='cc="'$CC'" cxx="'$CXX'" is_debug=false extra_cflags=["'"$CXXFLAGS_ARR"'","-DIS_FUZZING"] skia_use_system_freetype2=false skia_use_fontconfig=false skia_enable_gpu=false extra_ldflags=["-L'$LIB_FUZZING_ENGINE'", "'"$CXXFLAGS_ARR"'"]'
+$SRC/depot_tools/gn gen out/Fuzz --args='cc="'$CC'" cxx="'$CXX'" is_debug=false extra_cflags=["'"$CXXFLAGS_ARR"'","-DIS_FUZZING"] skia_use_system_freetype2=false skia_use_fontconfig=false skia_enable_gpu=false extra_ldflags=["-lFuzzingEngine", "'"$CXXFLAGS_ARR"'"]'
 
 
 
-$SRC/depot_tools/ninja -C out/Fuzz fuzz
\ No newline at end of file
+$SRC/depot_tools/ninja -C out/Fuzz fuzz_region_deserialize
+
+cp out/Fuzz/fuzz_region_deserialize $OUT/fuzz_region_deserialize
```
Comment (inferno-chromium) [inline projects/skia/build.sh]: Can you remove the fuzz_ from the fuzz target prefix. We need .options file and fuzz target to have same name.

Comment (kjlubick) [inline projects/skia/build.sh]: Done.

File: projects/skia/build.sh (hunk 4 above)
Comment (kjlubick) [inline projects/skia/build.sh]: Done.

File: projects/skia/build.sh (hunk 3 above)
Comment (kjlubick) [inline projects/skia/build.sh]: Done.

File: projects/skia/build.sh (hunk 2 above)
Comment (kjlubick) [inline projects/skia/build.sh]: Done.

**Pull request merged**